### ISSUE TITLE
Document that out-parameters used with verification in KDFs should be zeroized

### ISSUE THREAD

**Issue by brycx** [brycx/orion#178]
The `verify()` functions offered in `hazardous::kdf` require an out-parameter in order to support `no_std`. During verification, some amount of bytes are derived from a secret, which are placed into the out-parameter. These are then used to compare with an expected value.

Using the `derive_key()` functions, it is obvious that the out-parameter will contain the secret key, but it may not be obvious when using `verify()`. Therefor, the security section of documentation for all `hazardous::kdf` modules should mention that, sensitive values are placed into the out-parameter, but Orion does not zeroize them. So users must make sure they do if needed.

Example:
```
[..]
- When `verify()`ing, the `dst_out` parameter should always be zeroed out after finishing the operation.
```

However, for similar things, Orion usually offers utility functions such as `secure_rand_bytes` or `secure_cmp`. The question is whether Orion should offer a zeroization method (wrapper around `zeroize` which Orion uses internally) or simply point users to the [`zeroize` crate](https://crates.io/crates/zeroize)?

**Comment by brycx** [comment]
Only for `v016`, after #181.

**Comment by brycx** [comment]
See the update for #179 (`hkdf::verify()` will be removed).

Documentation of zeroizing out-parameters still seems applicable, in terms of PBKDF2 and Argon2i. In that case, the verification happens in the low-level API, and is expected to be used for passwords or similar. These are not necessarily sensitive values when copied to `dst_out` (hashes of password), so I believe it would be sensible to simply refer to the `zeroize` crate instead of Orion providing the functionality. As long as, the user is made aware that the derived material is copied thereto.

**Comment by vlmutolo** [comment]
For the vast majority of use cases, I think that orion users will consider `dst_out` to be more or less "safe", and you shouldn't worry too much about providing routines to zero it out.

I guess I can imagine a situation where someone maybe has reason to heap-allocate the `dst_out` buffer, so now the non-zeroed information is long-lived, and maybe potentially more vulnerable to compromise. Anyone in a situation where they're using the `hazardous` module, and who knows enough to care about what portions of memory may be compromised with an attack that has found some way of snooping on private memory, is likely to be okay with using the zeroize crate themselves. Documenting that `dst_out` has potentially sensitive information in it is all you need to do.

**Comment by brycx** [comment]
> I think that orion users will consider dst_out to be more or less "safe", and you shouldn't worry too much about providing routines to zero it out.

This is a good point. Especially because we don't give any guarantees that all sensitive memory is wiped. You raise a great point about the expectation of the user. I forgot that `hazardous` implies an "advanced" user, which in turn seems reasonable to assume that they'd know how to and do use a crate like `zeroize`. 

Thanks so much for taking a look @vlmutolo!

**Comment by brycx** [comment]
Fixed in #187
